From 301ac38e30138b2b337391a64d0770ca6a6eb468 Mon Sep 17 00:00:00 2001
From: Justin Bronder <jsbronder@gmail.com>
Date: Sat, 25 Jun 2011 21:07:41 -0400
Subject: [PATCH] disable automagic doc building

Instead of unconditionally building docs if doxygen is found, only build
at the request of the user.
---
 configure.ac |   23 +++++++++++++++--------
 1 files changed, 15 insertions(+), 8 deletions(-)

diff --git a/configure.ac b/configure.ac
index 8624a5f..2999094 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1323,8 +1323,6 @@ dnl
 if test "$build_drmaa" = "yes" ;then
 AC_MSG_NOTICE([Configuring DRMAA ...])
 AC_PATH_PROG(GPERF, "gperf")
-AX_PROG_DOT
-AC_PATH_PROG(DOXYGEN, "doxygen", "none")
 AC_C_BIGENDIAN_CROSS
 ACX_PTHREAD(,[AC_MSG_ERROR([POSIX threads library is required by DRMAA.])])
 
@@ -1368,12 +1366,21 @@ AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <stdarg.h>]],
   AC_MSG_RESULT([no])
 ])
 
-  build_drmaa_docs=yes
-  RPM_AC_OPTS="$RPM_AC_OPTS --with drmaa"
-else  
-  DOXYGEN=none
-  build_drmaa_docs=no
-  RPM_AC_OPTS="$RPM_AC_OPTS --without drmaa"
+AC_ARG_ENABLE([apidocs],
+    AS_HELP_STRING([--enable-apidocs],[Build DRMAA documentation])
+)
+AS_IF([test "x$enable_apidocs" = "xyes"],
+    [ AX_PROG_DOT
+      AC_PATH_PROG(DOXYGEN, "doxygen", "none")
+      RPM_AC_OPTS="$RPM_AC_OPTS --with drmaa"
+      build_drmaa_docs=yes ],
+    [ build_drmaa_docs=no
+      DOXYGEN=none ]
+)
+else
+    DOXYGEN="none"
+    build_drmaa_docs=no
+    RPM_AC_OPTS="$RPM_AC_OPTS --without drmaa"
 fi
 AM_CONDITIONAL(DRMAA_DOCS, [test "$DOXYGEN" != "none"])
 AC_SUBST(build_drmaa_docs)
-- 
1.7.3.4

